apps/web/components/dashboard/tags/AllTagsView.tsx

```diff
@@ -54,48 +57,128 @@ function DeleteAllUnusedTags({ numUnusedTags }: { numUnusedTags: number }) {
     </ActionConfirmingDialog>
   );
 }
-
-function TagPill({
-  id,
-  name,
-  count,
-}: {
-  id: string;
-  name: string;
-  count: number;
-}) {
-  return (
-    <div className="group relative flex">
-      <Link
-        className="flex gap-2 rounded-md border border-border bg-background px-2 py-1 text-foreground hover:bg-foreground hover:text-background"
-        href={`/dashboard/tags/${id}`}
-      >
-        {name} <Separator orientation="vertical" /> {count}
-      </Link>
-
-      <DeleteTagConfirmationDialog tag={{ name, id }}>
-        <Button
-          size="none"
-          variant="secondary"
-          className="-translate-1/2 absolute -right-1 -top-1 hidden rounded-full group-hover:block"
-        >
-          <X className="size-3" />
-        </Button>
-      </DeleteTagConfirmationDialog>
-    </div>
-  );
+interface DragState {
+  dragId: string | null;
+  dragOverId: string | null;
+  x: number;
+  y: number;
```

Comment:
Let's add a comment here and give them a bit more descriptive name?

```suggestion
// The position of the elements being dragged such that on drag over, we can revert the position.
  initialX: number;
  initialY: number;
```